File: src/main/java/io/odxproxy/client/OdxProxyClientInfo.java
package io.odxproxy.client;

import io.odxproxy.model.OdxInstanceInfo;

public class OdxProxyClientInfo {
    public final OdxInstanceInfo instance;
    public final String odxApiKey;
    public final String gatewayUrl;

    public OdxProxyClientInfo(OdxInstanceInfo instance, String odxApiKey, String gatewayUrl) {
        this.instance = instance;
        this.odxApiKey = odxApiKey;
        this.gatewayUrl = gatewayUrl;
    }
     public OdxProxyClientInfo(OdxInstanceInfo instance, String odxApiKey) {
        this(instance, odxApiKey, "https://gateway.odxproxy.io");
    }
}

File: src/main/kotlin/io/odxproxy/client/OdxProxyClientInfo.kt
package io.odxproxy.client

import io.odxproxy.model.OdxInstanceInfo

public data class OdxProxyClientInfo @JvmOverloads public constructor(
    public val instance: OdxInstanceInfo,
    public val odxApiKey: String,
    public val gatewayUrl: String = "https://gateway.odxproxy.io"
)